staging/src/k8s.io/apiserver/pkg/storage/cacher.go

```diff
@@ -888,6 +888,15 @@ func (c *cacheWatcher) sendWatchCacheEvent(event *watchCacheEvent) {
 		watchEvent = watch.Event{Type: watch.Deleted, Object: oldObj}
 	}
 
+	// SWAT Event lost: print out resourceVersion, EventType and event object name
+	// if event is for a pod or a node
+	meta, err := meta.Accessor(watchEvent.Object)
+	if err != nil {
+		return
+	}
+	fmt.Printf("SWAT,cacher/sendWatchCacheEvent,%s,%s,%s\n",
+		watchEvent.Type, meta.GetName(), meta.GetResourceVersion())
```

Comment:
It looks like most (if not all) of the `runtime.Object` carried by these events don't set the `TypeMeta`. As a result `watchEvent.Object.GetObjectKind().GroupVersionKind()` returns empty GVK. Don't know why that comes to happen (maybe apiserver doesn't need to know gvk when it's serving objects from etcd to clients). 

You could use `reflect.TypeOf()` to identify the Go type for now.

Reply:
OK, thanks. I will do that.